### PR TITLE
Update Style Guide for naming of Models and Layers

### PR DESCRIPTION
**Pull request by sampathweb**

### CODE REVIEW

File: STYLE_GUIDE.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ For the most part, following our code style is very simple, we just use
 [black](https://github.com/psf/black) to format code. See our
 [Contributing Guide](CONTRIBUTING.md) for how to run our formatting scripts.
 
+## Naming of Layers and Models
+
+Capitalize all abbreviations, e.g. LSTM not Lstm, KLDivergence not KlDivergence, GPT2, XLMRoberta, etc.
+
+When a specific abbreviation is very common and is pronounceable (acronym), consider it as a standalone word, e.g. Bert, Deberta, etc. In this case, "Bert" is considered as a common noun and not an abbreviation anymore.
```
Comment (mattdangerw) [inline STYLE_GUIDE.md:13]: Can you break up this line at 80 characters?

File: STYLE_GUIDE.md [hunk 2]
```diff
@@ -6,6 +6,12 @@ For the most part, following our code style is very simple, we just use
 [black](https://github.com/psf/black) to format code. See our
 [Contributing Guide](CONTRIBUTING.md) for how to run our formatting scripts.
 
+## Naming of Layers and Models
+
+Capitalize all abbreviations, e.g. LSTM not Lstm, KLDivergence not KlDivergence, GPT2, XLMRoberta, etc.
```
Comment (mattdangerw) [inline STYLE_GUIDE.md:11]: Capitalize all abbreviations -> Capitalize all acronyms?

File: STYLE_GUIDE.md [hunk 3]
```diff
@@ -6,6 +6,12 @@ For the most part, following our code style is very simple, we just use
...
+
+When a specific abbreviation is very common and is pronounceable (acronym), consider it as a standalone word, e.g. Bert, Deberta, etc. In this case, "Bert" is considered as a common noun and not an abbreviation anymore.
+
```
Comment (mattdangerw) [inline STYLE_GUIDE.md:14]: Let's also mention filenames here. E.g.

Files should be named with snake case, and an acronym should be consider a single "segment." For example `XLMRoberta` would map to an `xlm_roberta.py` filename.

**Review by mattdangerw — CHANGES_REQUESTED** [review]
Thanks!

File: STYLE_GUIDE.md (hunk 1 above)
Comment (sampathweb) [reply STYLE_GUIDE.md]: Done

File: STYLE_GUIDE.md (hunk 2 above)
Comment (sampathweb) [reply STYLE_GUIDE.md]: Thanks. Done

File: STYLE_GUIDE.md (hunk 3 above)
Comment (sampathweb) [reply STYLE_GUIDE.md:20]: Thanks for the additional details.  Updated to include guideline on filenames.

Comment (mattdangerw) [reply STYLE_GUIDE.md:20]: Thank you!

**Review by mattdangerw — APPROVED** [review]

**Pull request merged**